### ISSUE TITLE
Non-iterable error on attr.fields iterator

### ISSUE THREAD

**Issue by sveinse** [PyCQA/pylint#1884]
```
$ pylint --version output
No config file found, using default configuration
pylint 1.8.2,
astroid 1.6.1
Python 3.6.4 (v3.6.4:d48eceb, Dec 19 2017, 06:04:45) [MSC v.1900 32 bit (Intel)]
```

Code:
``` python
import attr

@attr.s
class A(object):
    x = attr.ib()
    y = attr.ib(default=42)

    def reset(self):
        for field in attr.fields(type(self)):
            if field.default is not attr.NOTHING:
                setattr(self, field.name, field.default)
```

This fails with 
```
E:  9,21: Non-iterable value attr.fields(type(self)) is used in an iterating context (not-an-iterable)
```

**Comment by brycepg** [comment]
Looks like the indirect way attrs 'builds' classes causes astroid to not find `__attrs_attrs__` field for the attr class. This culminates in the inference of `getattr(cls, "__attrs_attrs__", None)` to return `None` (which causes the iteration error).

A solution to this issue would be to patch the attrs brain in astroid to have a locals value for this field.

**Comment by brycepg** [comment]
Thanks for the report!

I've fixed this issue for in astroid.
